src/sentry/uptime/endpoints/project_uptime_alert_checks_index.py

```diff
@@ -193,6 +198,17 @@ def _format_row(
         for col_idx, col_name in enumerate(column_names):
             result = column_values[col_idx].results[row_idx]
             row_dict.update(self._extract_field_value(result, col_name, uptime_subscription))
+
+        # XXX: Translate the status from `failed` to `failed_incident` when the
+        # check is part of an incident.
+        #
+        # TODO(epurkhiseer): In the future this should likely be part of the
+        # serializer that handles translating the snuba data into the expected
+        # data for the frontend.
+        in_incident = row_dict["incident_status"] == IncidentStatus.IN_INCIDENT.value
+        if row_dict["check_status"] == "failure" and in_incident:
+            row_dict["check_status"] = "failure_incident"
```

Comment:
Do we also need to produce `failure_no_incident`?

Reply:
nah, that is just `failure`